### ISSUE TITLE
Added option to sorted/collapse to show entities under a certain threshold expanded

### ISSUE THREAD

**Issue by jwelter1971** [maxwroc/battery-state-card#44]
**Is your feature request related to a problem? Please describe.**
Not a problem, just a feature expansion

**Describe the solution you'd like**
When specifying sort and collapse you specify a fixed # of entities that are expanded initially.  It would be nice to have any entities under a specified threshold expanded initially.  So for example devices under XX% battery threshold.

**Additional context**
N/A

**Comment by maxwroc** [comment]
Interesting idea but i wonder what should happen if you specify threshold but there are no batteries which are under? Should we show just the expand button (this would look little bit odd I think)? or should we fallback to some default number of visible entities?

**Comment by jwelter1971** [comment]
Good question.  Perhaps it should work with "collapse" and the default would be to do what was specified, but over-ride this and expand that to show any under the threshold?

**Comment by maxwroc** [comment]
Ok make sense but I would like to understand the actual benefit of having such feature. 

Usually you're interested in devices having the lowest battery level (to remember to replace it), right? Why you want to see all below certain threshold? (just wondering)

**Comment by torkelsson** [comment]
With 50+ sensors/remotes etc with batteries, the number that needs battery replacement (i.e those below a certain threshold) can vary a lot from time to time, from 1 to 20 (there are usually at least some unit that needs a new battery all the time), and there is usually also at least one sensor that shows up as unavailable. I still want everything that may need battery replacement at some time in one card though. 

Normally I'm only interested in those that needs to be replaced (or are unavailable) but sometimes, when running out of batteries, I'm also interested in knowing which are in "yellow state" so I can buy more batteries. And of course sometimes I'm interested in seeing all units. 

If possible it would actually be really valuable to be able to do multifoldning based on the thresholds (and state). I.e fold by the threshold for "red", "yellow", "green", and states "unavailable", "charging" etc. 

I'm not sure that just showing the expand button would look that weird. If it, does perhaps you can add an (optional) title that can be shown for each "level"?

**Comment by maxwroc** [comment]
Hey guys, sorry I was not replying here right away because I was thinking about best way of handling mentioned problems.

I can clearly see now problem of having too many entities and that they can flood the card/UI. So indeed it make sense to have such option to show entities which are below some threshold (to avoid need of expanding the the rest).

BTW please note that in v1.3.0 I have added filtering options which can be used to remove batteries with 100% level (or above certain threshold). You may find it useful.

@torkelsson Nice idea of having expandable groups. I will think about this more.

I'm considering as well adding a glance card type of view (look below). You would be able to show entities in little bit more compact way. But in this case you have very small area to display the name (I have some ideas though how to do it - better than on image below).

Note: below screenshot is not the exact proposal. It was posted by some user somewhere on the github
![image](https://user-images.githubusercontent.com/8268674/82564802-4c2f4f00-9b71-11ea-81de-1d4d1b164426.png)

**Comment by maxwroc** [comment]
@jwelter1971 , @torkelsson 
@bieniu (as I saw you have a lot of battery cards in your configuration)

Hey just a small update about this idea guys. I'm getting closer to the approach I wanted to achieve and wanted to hear your feedback (if there is any). 

I have noticed that a lot of people groups the sensors by place where they are located or the type of them. So I thought that if people have the defined groups in HA already why not to use them. You will be able to add entity groups instead of single entities now.

In `collapse` node, instead of just the number, you will be able to define collapsible groups. For all of them you can specify the following options:
```yaml
name: "My group" #Name which will appear on the left side of the chevron
icon: mdi:temp #Icon to show for the group
min: 20 #Minimum level of battery
max: 90 #Maximum level of battery
group_id: group.temperature_sensors #Group ID - must be matching one of the groups used in entities collection
entities: #List of entities for this group (if you don't want to use HA groups)
  - sensor.bedroom_temperature_sensor
  - sensor.main_door_sensor
```
The `min`, `max`, `group_id` and `entities` are conditions which, if set, all needs to be met by battery to be assigned to the particular group.

All batteries which won't fit in any of these groups will be shown as usual on the beginning (not grouped). This way you will be able to show on top devices which require battery replacement and if you want to check what is the state for the rest of devices you will be able to expand sections.

In the `name` property you can use keywords like `{min}`, `{max}`, `{count}`.

Just bear in mind that it is not the final look:
![image](https://user-images.githubusercontent.com/8268674/83577290-bd84df80-a52b-11ea-9911-c4d697728937.png)


I plan to add spacing to align "chevron" row with other entities (it will become same size as regular entity row). Currently icons are missing and I want to use the HA group icon if it is defined, unless the override from my card is not specified. 

I'm thinking about a way to apply some colors to the group icon but I'm not sure yet how it will work (if there will be a separate config option to specify whether it should be min/max/avg color for given group or something else).

I hope you will like this idea and I look forward for your feedback.

**Comment by maxwroc** [comment]
Released in v1.4.0 as promised
